languages/scala/recursive_descent/Parser_scala_recursive_descent.ml

```diff
@@ -3387,23 +3419,34 @@ let templateStatSeq ~isPre in_ : self_type option * block =
           (noSelfType, Some (E first)))
     else (noSelfType, None)
   in
-  let xs = templateStats in_ in
+  let xs = templateStats startTok in_ in
   (self, Option.to_list firstOpt @ xs)
 
 (** {{{
- *  TemplateBody ::= [nl] `{` TemplateStatSeq `}`
+ *  TemplateBody ::= [nl] :<<< `{` TemplateStatSeq `}` >>>
  *  }}}
  * @param isPre specifies whether in early initializer (true) or not (false)
 *)
 
 let templateBody ~isPre in_ : template_body =
-  (* ast: self, EmptyTree.asList *)
-  inBraces (templateStatSeq ~isPre) in_
+  match in_.token with
+  | LBRACE _ ->
+      (* ast: self, EmptyTree.asList *)
+      inBraces (templateStatSeq ~isPre) in_
+  | _ ->
```

Comment:
Should we check that it's actually a Colon here?

Reply:
the `accept` will throw an error anyways, so it's the same as throwing an explicit new error in the third wildcard case. by convention, in other parts of this parser where we narrow down to a single case, we generally wildcard it and just enforce constraints via exceptions